code/modules/mob/living/basic/lavaland/goldgrub/goldgrub_abilities.dm

```diff
@@ -0,0 +1,53 @@
+/datum/action/cooldown/mob_cooldown/spit_ore
+	name = "Spit Ore"
+	desc = "Vomit out all of your consumed ores."
+	click_to_activate = FALSE
+	cooldown_time = 5 SECONDS
+
+/datum/action/cooldown/mob_cooldown/spit_ore/Activate()
+	var/mob/living/basic/mining/goldgrub/grub_owner = owner
+	grub_owner.barf_contents()
+	StartCooldown()
+	return TRUE
```

Comment:
The original action had a check for `is_burrowed`, you should probably put that in `IsAvailable`